apps/giveaways/grpc_impl/grpc_impl.go

```diff
@@ -0,0 +1,66 @@
+package grpc_impl
+
+import (
+	"context"
+	"errors"
+	"github.com/samber/lo"
+	model "github.com/satont/tsuwari/libs/gomodels"
+	"gorm.io/gorm"
+	"time"
+
+	"github.com/satont/tsuwari/apps/giveaways/internal/types/services"
+	"github.com/satont/tsuwari/libs/grpc/generated/giveaways"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+type GiveawaysGrpcServer struct {
+	giveaways.UnimplementedGiveawaysServer
+
+	services *services.Services
+}
+
+func NewServer() *GiveawaysGrpcServer {
+	return &GiveawaysGrpcServer{}
+}
+
+func (server *GiveawaysGrpcServer) HandleChatMessage(
+	ctx context.Context,
+	data *giveaways.HandleChatMessageRequest,
+) (*emptypb.Empty, error) {
+	/*
+		Firstly, we need to check that some giveaway is running for current channel
+	*/
+	giveaway := &model.ChannelGiveaway{}
+	err := server.services.Gorm.WithContext(ctx).
+		Where(`"channel_id" = ? AND "start_at" > ? AND "end_at < ?"`, data.Channel.Id, time.Now().String(), time.Now().String()).
```

Comment:
Capsulating via `"` not needed, if you using `_` in variables instead of `camelCase`, but thats not important.